File: Data Structures & Algorithms/top-k-elements-in-list/submission-6.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        let count=new Map()
        for(const num of nums){
            count.set(num,(count.get(num)||0)+1)
        }
        let occ=new Array(nums.length+1).fill(null)
        for(const [key,value] of count.entries()){
            if(!occ[value]) occ[value]=[]
            occ[value].push(key)
        }
        let result=[]
        for(let num=occ.length-1;num>=0;num--){
            if(k===0) break
            if(occ[num]){
                result.push(...occ[num])
                k-=occ[num].length
            }
        }
        return result
    }
}

File: Data Structures & Algorithms/three-integer-sum/submission-6.py
class Solution:
    def threeSum(self, nums: List[int]) -> List[List[int]]:
        result=set()
        nums.sort()
        for i in range(len(nums)-2):
          a,b=i+1,len(nums)-1
          while a<b:
            total = nums[i] + nums[a] + nums[b]
            if not total:
              result.add((nums[i],nums[a],nums[b]))
            if total>0:
              b-=1
            else:
              a+=1
        return ([list(i) for i in result])
          
